### ISSUE TITLE
Comment form: design change suggestions and accessibility improvements

### ISSUE THREAD

**Issue by Nyiriland** [Automattic/wp-calypso#64913]
### 1 - Form field padding inconsistent
I noticed on the default dotcom comment form, the textarea container (`#comment-form-comment`) has smaller padding on the bottom than the top and sides. Does this have a purpose? Screencap below, where you can see how apparent the uneven padding is when full of text:

<img width="725" alt="image" src="https://user-images.githubusercontent.com/3835008/175380374-395c1c20-68e2-4e4b-bab0-47f76b53f79b.png">

### 2 - Text sizing not great for readability

As I started writing the above issue, I also realised how out of date the form field is, a11y-speaking:

- Line height of textarea could be increased for readability (1.4 is browser standard)
- According to WCAG 2.1 standards, font sizes ideally should be minimum 16px for body text, and never smaller than 12px.

<img width="793" alt="image" src="https://user-images.githubusercontent.com/3835008/175391557-9dd9aa01-b43e-4ff4-a6b9-21dd46510fad.png">

### 3 - Proposed improvements for a11y

The below screenshot shows the same form, with adjustments added for easier readability:

<img width="762" alt="image" src="https://user-images.githubusercontent.com/3835008/175386102-7e3c6ebe-1d59-4ff9-95fa-cec2ee617dc4.png">

- Textarea font size increased to 16px, line height 1.4
- Padding around textarea (`#comment-form-comment`) increased to 16px (from 10)
- instructions and input field placeholder font increased to 14px (from 12px)
- padding around inputs increased to 6px (from 4px/5px)
- input field helper text (the stuff in parentheses) increased to 13px (from 11px)
- for maximum accessibility, the placeholder text should actually be visible outside the form field at all times... but that would perhaps involve a larger redesign.
- I didn't change this in the above screenshot, but the input placeholder text [should also be darkened to meet accessible contrast ratios](https://accessibleweb.com/question-answer/does-placeholder-text-need-to-meet-color-contrast-requirements/).

### Steps to reproduce

Place Comment Form block into any theme.

### What you expected to happen

See above.

### What actually happened

See above.

### Simple, Atomic or both?

Simple, Atomic

### Theme-specific issue?

The styling of the comment form is theme-agnostic. I have not personally tried this on Atomic, but I assume the results would be the same.

### Browser, operating system and other notes

I'm on a mac, looking at Chrome.

### Reproducibility

Consistent

### Severity

All

### Available workarounds?

No but the platform is still usable

**Comment by Robertght** [comment]
I'm not sure who is in charge of this, but from my findings it could be more suitable for Jetpack. I can see the styling comes from /mu-plugins/highlander-comments. Is Jetpack territory? cc @jeherve 

📌 HOUSEKEEPING
- Labels ✅ 
- Priority ✅ 
- Replicable on Core - Yes/No

📌 SCRUBBING
- Tested on P2 - no
- Tested on Simple ✅ 
- Tested on AT ✅ 
- Tested on Self-hosted ✅ 

📌 FINDINGS/SCREENSHOTS/VIDEO
- I can replicate the above.
- Checking to see where this could be placed better

📌 ACTIONS
- Marked as Triaged for Quality Squad review

**Comment by jeherve** [comment]
> I'm not sure who is in charge of this

No one at the moment. The comment form could definitely use some design love, its design hasn't changed at all in the past 10 years.

> I can see the styling comes from /mu-plugins/highlander-comments. 

That's correct. That file is on WordPress.com, since the comment form is the same on WordPress.com simple and on Jetpack environments.

Given that we do not currently actively maintain and improve that part of WordPress.com, I'm afraid I cannot suggest a team that could help with this. I would recommend keeping this issue around though; when we next iterate on the comment form feature, we can refer to this issue for ideas.

**Comment by Nyiriland** [comment]
> The comment form could definitely use some design love, its design hasn't changed at all in the past 10 years.

Agreed! Since the form has very opinionated styling (ie: all fonts are Arial), I wonder if modernizing it should involve allowing the form (or at least the textarea and submit button) to inherit styling from the theme? It curiously already does when previewing the form in the editor, but on the front end, default styles get applied. Here's a sample of the visual disparity:

<img width="793" alt="image" src="https://user-images.githubusercontent.com/3835008/176907673-c7c7a88f-086e-4217-8552-1804b2f90725.png">

**Comment by jeherve** [comment]
> I wonder if modernizing it should involve allowing the form (or at least the textarea and submit button) to inherit styling from the theme? It curiously already does when previewing the form in the editor, but on the front end, default styles get applied. Here's a sample of the visual disparity

Ideally yes (this is tracked in https://github.com/Automattic/jetpack/issues/13277 ), but that's not really possible with the current implementation of the feature; the comment form is added in an iFrame, and consequently lives on its own on WordPress.com and will not be impacted by the theme's CSS. 
This is why you see that difference between the form preview (using WordPress' default comment system, living on the site and impacted by the theme's CSS) and the form on the frontend (using the WordPress.com comment form, in an iFrame).

**Comment by mpkelly** [comment]
I am going to take a look a this one.

**Comment by daledupreez** [comment]
Most of these issues for WordPress.com Simple sites have been addressed by @mpkelly in D85615-code which was deployed a few minutes ago, however, we realised after deploying that there are still some theme-dependent issues for WordPress.com Simple:
1. Some themes don't specify fonts for the comment area, and the fonts may be getting overridden in the comments section (this doesn't apply to themes that do specify fonts with enough CSS specificity to get precedence)
2. Some themes specify some CSS rules that result in unexpected spacing for checkboxes and/or labels

We still need to take a look at the behaviour for Atomic and Jetpack sites.

**Comment by daledupreez** [comment]
We took on a good deal of additional in D85984-code, and finally got that deployed today after making sure we had sufficient visibility for the changes: p1HpG7-hu9-p2

We still have some more work to do so that the comment form aligns with the theme more closely (at least on WordPress.com Simple sites), and we plan to tackle that work as part of https://github.com/Automattic/wp-calypso/issues/67212.

I am going to close this issue out, as I think it may be better to tackle smaller improvements moving forward, but please feel free to re-open the issue if you think we have some broader improvements that still need to be made.
